### PR TITLE
Fetch or fallback with deprecations

### PR DESCRIPTION
**Pull request by manuelpuyol**
Adds a simple way of declaring deprecated values using `fetch_or_fallback`. It will now receive a `deprecated_values:` argument with a list of deprecated values. It will use them as possible options, but will add a deprecation warning (not in production) when used.

Deprecates `orange` and `purple` schemes for LabelComponent to showcase how to use it properly.

Added `mocha` to help us with mocks in tests

### CODE REVIEW

File: app/components/primer/label_component.rb [hunk 1]
```diff
@@ -37,23 +38,18 @@ class LabelComponent < Primer::Component
...
-    #   <%= render(Primer::LabelComponent.new(title: "Label: Label", scheme: :orange)) { "Orange" } %>
-    #   <%= render(Primer::LabelComponent.new(title: "Label: Label", scheme: :purple)) { "Purple" } %>
-    #
     # @param title [String] `title` attribute for the component element.
     # @param scheme [Symbol] <%= one_of(Primer::LabelComponent::SCHEME_MAPPINGS.keys) %>
     # @param variant [Symbol] <%= one_of(Primer::LabelComponent::VARIANT_OPTIONS) %>
     # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
     def initialize(title:, scheme: nil, variant: nil, **system_arguments)
       @system_arguments = system_arguments
-      @system_arguments[:bg] = :blue if scheme.nil?
```
Comment (manuelpuyol) [inline app/components/primer/label_component.rb:50]: this class has no effect in `Label`

Comment (joelhawksley) [reply app/components/primer/label_component.rb:46]: cc @simurai any idea how this ended up here?

Comment (manuelpuyol) [reply app/components/primer/label_component.rb:46]: This was needed pre-PrimerCSS v15

Comment (simurai) [reply app/components/primer/label_component.rb:46]: 👍  Yeah, I think `.Label--blue` [changed](https://primer-css-git-mkt-color-modes-docs.primer.vercel.app/css/support/v16-migration#labels) `.Label--info`.

File: test/test_helper.rb [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ENV["RAILS_ENV"] = "test"
 
 require "minitest/autorun"
+require "mocha/minitest"
```
Comment (srt32) [inline test/test_helper.rb:6]: Could you remind me what we get from this dep?

**Review by srt32 — APPROVED** [review]

**Pull request merged**